doc/manual/manual.tex

```diff
@@ -5,183 +5,241 @@
+    ale.setInt("random_seed", 123);
+    ale.loadROM(argv[1]);
+
+    ActionVect legal_actions = ale.getLegalActionSet();
+    
+    float totalReward = 0;
+    while (!ale.game_over()) {
+        Action a = legal_actions[rand() % legal_actions.size()];    
+        float reward = ale.act(a);
+            totalReward += reward;
+        }
+        cout << "The episode ended with score: " << totalReward << endl;
+    }
+    return 0;
 }
+\end{verbatim}
 
-\subsection{RL-Glue Interface Arguments}
+Recall this is a very simple example using the ALE. It is also important to remember to use the flag 
+\verb+-L+ to link the ALE library. Also, to run the agent, you may need to add ALE to your library
+path:
 
-\small{
 \begin{verbatim}
-  -send_rgb <true|false> -- if true, RGB values are sent for each pixel
-      instead of the pallette index values
-    default: false
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/path/to/ale_0_5
 \end{verbatim}
-}
 
-\section{Shared Library Interface}\label{sec:shared_library_interface}
-
-The shared library interface allows agents to directly access ALE via a class called
-\verb+ALEInterface+, defined in \verb+ale_interface.hpp+. 
-
-\subsection{Sample Agent}
-
-Example source code can be found under
+or under Mac OS X,
 
 \begin{verbatim}
-ale_0_4/doc/examples/sharedLibraryInterfaceExample.cpp
+export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:/path/to/ale_0_5
 \end{verbatim}
 
-Assuming you installed ALE under \verb+/home/marc/ale_0_4+, the shared library agent can be 
-compiled with the following command: 
+As a final remark, a complete list of functions available in the class \verb+ALEInterface+ is 
+presented in Section~\ref{sec:functions}.
 
-\begin{verbatim}
-g++ -I/home/marc/ale_0_4/src -L/home/marc/ale_0_4 
-  -lale -lz sharedLibraryInterfaceExample.cpp 
-\end{verbatim}
+\section{Implementing an agent in Python (shared libraries)}\label{sec:python_interface}
 
-or alternatively by appropriately setting paths in \verb+ale_0_4/doc/examples/Makefile.sharedlibrary+ before running 
+To use the Python interface it is necessary to install it after the ALE was compiled. If one has 
+root access to the machine it is enough to run:\\
 
-\begin{verbatim}
-ale_0_4/doc/examples> make sharedLibraryAgent
-\end{verbatim}
+\verb+pip install .+\\
 
-To run the agent, you may need to add ALE to your library path:
+otherwise, if he does not have root access, the following should work:\\
 
-\begin{verbatim}
-export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/marc/ale_0_4
-\end{verbatim}
+\verb+pip install --user .+\\
 
-or under Mac OS X,
+Then, to use it in a Python code it is enough to import it properly, for example, as in the 
+example below (\verb+from ale_python_interface import ALEInterface+). A code containing
+an example is available at \verb+doc/examples/python_example.py+.
+
+All the functions are the same as in the C++ interface. Therefore, it is pointless to discuss all
+functions again. A complete example in Python is below:
```

Comment:
The functions are similar to the C++ interface. They are not identical IIRC. 
Instead of saying it is pointless, maybe we can just say that we omit the line-by-line description.